pyspectral/rayleigh.py

```diff
@@ -228,16 +230,29 @@ def get_reflectance_lut(self):
             self._azid_coord = lut_vars[2]
             self._satz_sec_coord = lut_vars[3]
             self._sunz_sec_coord = lut_vars[4]
-        return self._rayl, self._wvl_coord, self._azid_coord,\
-            self._satz_sec_coord, self._sunz_sec_coord
+        return self._rayl, self._wvl_coord, self._azid_coord, \
+               self._satz_sec_coord, self._sunz_sec_coord
 
     @staticmethod
     def _do_interp(minterp, sunzsec, azidiff, satzsec):
         interp_points2 = np.vstack((sunzsec.ravel(), 180 - azidiff.ravel(), satzsec.ravel()))
         res = minterp(interp_points2)
         return res.reshape(sunzsec.shape)
 
-    def get_reflectance(self, sun_zenith, sat_zenith, azidiff, bandname, redband=None):
+    @staticmethod
+    def _do_reduce_effect(zenith, rayref, thresh_ang, maxzen, strength):
+        """Reduce the Rayleigh correction amount at high zenith angles.
+        This linearly scales the Rayleigh reflectance, `rayref`, for solar or satellite zenith angles, `zenith`,
+        above a threshold angle, `thresh_ang`. Between `thresh_ang` and `maxzen` the Rayleigh reflectance will
+        be linearly scaled, from one at `thresh_ang` to zero at `maxzen`.
+        """
+        factor = 1. - strength * where(zenith < thresh_ang, 0, (zenith - thresh_ang) / (maxzen - thresh_ang))
+        # For low zenith factor can be greater than one, so we need to clip it into a sensible range.
+        factor = clip(factor, 0, 1)
+        return rayref * factor
+
+    def get_reflectance(self, sun_zenith, sat_zenith, azidiff, bandname, redband=None,
```

Comment:
Nice work @simonrp84 But the `get_reflectance` method is getting increasingly complex and have too many input arguments. But that was already bad before. Could the effect reduction not be applied after, as a separate function on the standard correction?